transform/mattermost-analytics/models/staging/stripe/stg_stripe__products.sql

```diff
@@ -0,0 +1,39 @@
+with source as (
+
+    select * from {{ source('stripe_raw', 'products') }}
+
+),
+
+products as (
+
+    select
+        active,
+        attributes,
+        created,
+        description,
+        id,
+        images,
+        livemode,
+        metadata:"cws-product-family"::varchar as cws_product_family,
+        metadata:"cws-sku-name"::varchar as cws_sku_name,
+        metadata:"cws-allowable-addons"::varchar as cws_allowable_addons,
+        metadata,
```

Comment:
Do we need the metadata? Better to extract everything here.

Reply:
lets keep the metadata, we could extract everything, but if they add new fields or if they change the name of the JSON object which might lead to nulls because we need to update the fields, it will be easy to compare.

Reply:
Keeping the metadata and allowing downstream models to use them can introduce ambiguity. A typical scenario would be a downstream model extracting a field from `metadata` already extracted in the current staging model and giving it a different name than the one in the current model. 

The staging layer is also useful for standardizing naming and structure. If new fields are added in the `metadata` json field,  any renaming and handling must happen to the staging model. This approach also allows introducing tests to detect issues exactly like the ones described by @sri-byte. For example, a `not_null` test can be added on the extracted fields if we know that all rows must have a specific key in `metadata`.

Reply:
Removing metadata and adding only the columns that we need right now. In case any new field is needed from metadata we can always update the staging models.